src/Csv.php

```diff
@@ -379,32 +399,29 @@ public function parse($input = null, $offset = null, $limit = null, $conditions
             $input = $this->file;
         }
 
-        if (!empty($input)) {
-            if (!is_null($offset)) {
-                $this->offset = $offset;
-            }
+        if (empty($input)) {
+            // todo: but why true?
+            return true;
```

Comment:
You're right about the todo. The return value used to be `false` if `$this->data === false` - but that is weird, as a second, unsuccessful call to `parse` would still return `true`. So in short, it should return `false` here. I'll add a test for both cases when I merge it.